docs/walkthroughs/self-healing.md

```diff
@@ -0,0 +1,223 @@
+                        socket_address: { address: 127.0.0.1, port_value: 8080 }
+    listeners:
+      - name: http
+        value: |
+          name: http
+          # Changed listener port from 8081 to 5000
+          address: { socket_address: { address: 0.0.0.0, port_value: 5000 } }
+          filter_chains:
+            - filters:
+                - name: envoy.filters.network.http_connection_manager
+                  typed_config:
+                    "@type": type.googleapis.com/envoy.extensions.filters.network.http_connection_manager.v3.HttpConnectionManager
+                    stat_prefix: ingress_https
+                    route_config:
+                      name: local_route
+                      virtual_hosts:
+                        - name: any
+                          domains: ["*"]
+                          routes:
+                            [{ match: { prefix: "/" }, route: { cluster: "kuard" } }]
+                    http_filters:
+                      - name: envoy.filters.http.router
+EOF
+```
+
+After applying this EnvoyConfig, the `CacheState` of the EnvoyConfig resource will be `Rollback` because all the Envoy proxies consuming this config rejected the listener update and MARIN3R detected it and reverted the config to the lastest working one.
+
+```bash
+▶ kubectl get envoyconfig kuard
+NAME    NODE ID   ENVOY API   DESIRED VERSION   PUBLISHED VERSION   CACHE STATE
+kuard   kuard     v3          7966f8f4b7        7cbf9788cd          Rollback
+```
+
+We can check that the kuard app is still available at port 8081.
+
+```bash
+▶ curl http://localhost:8081 -v -o /dev/null --silent
+*   Trying ::1:8081...
+* Connected to localhost (::1) port 8081 (#0)
+> GET / HTTP/1.1
+> Host: localhost:8081
+> User-Agent: curl/7.76.1
+> Accept: */*
+>
+* Mark bundle as not supporting multiuse
+< HTTP/1.1 200 OK
+< content-length: 1592
+< content-type: text/html
+< date: Fri, 02 Jul 2021 14:22:36 GMT
+< x-envoy-upstream-service-time: 1
+< server: envoy
+<
+{ [1592 bytes data]
+* Connection #0 to host localhost left intact
+```
+
+Next time a config update is applied and accepted by all proxies, the `Rollback` status will go back to `InSync`.
+
+## **Conclusion**
+
+We have seen how MARIN3R tracks the status of Envoy resource updates and rollbacks the configuration when it sees that a given update is being rejected by all proxies. Its important to note several considerations about the self-healing functionality:
+
+- Failing configuration updates have an impact in Envoy performance because the xDS server will keep trying to send the update to the proxies and the proxies will keep trying to apply the update, failing in a loop. Even though this is mitigated by a backoff retry strategy in the xDS server side, the self-healing functionality avoids this error loop from going on forever. There is one case when this situation is not revertible though, which occurs when there is no previous correct config to rollback to.
```

Comment:
```suggestion
- Failing configuration updates have an impact on Envoy performance because the xDS server will keep trying to send the update to the proxies and the proxies will keep trying to apply the update, failing in a loop. Even though this is mitigated by a backoff retry strategy in the xDS server-side, the self-healing functionality avoids this error loop from going on forever. There is one case when this situation is not revertible though, which occurs when there is no previous correct config to roll back to.
```